homeassistant/components/tradfri/fan.py

```diff
@@ -145,10 +156,12 @@ async def async_turn_on(
 
         if percentage is not None:
             await self.async_set_percentage(percentage)
+            await self.coordinator.async_request_refresh()
```

Comment:
I'm thinking we shouldn't need to request a refresh at the end of the service calls since the observation callback should update state for us.

Reply:
Tried removing it and it still works so I'll push that change.